### PR TITLE
TRT update to 7.2 CUDA11

### PR DESCRIPTION
**Pull request by jayfurmanek**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/master/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/master/doc/)?
- [ ] Did you write any [tests](https://github.com/open-ce/open-ce/blob/master/tests/) to validate this change?  

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) requests changes, they must be addressed.

This has got to be one of the most complicated feedstocks.
 - x86_64 and ppc64le have different supported versions
 - NVIDIA doesn't make a 7.2 version available for ppc64le and CUDA10.2, so we'll likely have to drop that combo (or fall all the way back to 7.0 for CUDA 10.2)
 - the previous `memcpy` wrapper .so was moved to it's own recipe in this PR. This provides memcpy@GLIBC_2.14 for anything building with the libnvinfer.so and the anaconda toolchain, which uses a cos6 glibc (2.12)
 - another wrapper is needed - this time for `clock_gettime@GLIBC_2.17` in libnvinfer_plugins.so. So another recipe is added to create that
 - Most of the patches requires edits. Thankfully the pack-in packages (uff, graphsurgeon) are nice and clean now for TF2.
 - A new pack-in is included this time: onnx-graphsurgeon; this optionally wants onnx-runtime, which we don't yet include, but should at some point
 - the packed in samples are still pretty iffy at best, and have not been updated for TF2 (or fixed in any way)

### CODE REVIEW

File: libclock_gettime_wrapper/build.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+# *****************************************************************
+#
+# Licensed Materials - Property of IBM
+#
+# (C) Copyright IBM Corp. 2019,2020. All Rights Reserved.
```
Comment (npanpaliya) [inline libclock_gettime_wrapper/build.sh:6]: This should be just 2020.

File: libclock_gettime_wrapper/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  build:
+    - {{ compiler('c') }}
+
+about:
+  home: open-ce
+  license: apache-2
+  summary: summary
+  description: |
+    This is a wrapper package to map clock_gettime_GLIBC_PRIVATE to GLIBC_2.17
+  doc_url:
```
Comment (npanpaliya) [inline libclock_gettime_wrapper/meta.yaml:22]: Do we want to point to some reference doc for this?

File: libmemcpy_wrapper/build.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+# *****************************************************************
+#
+# Licensed Materials - Property of IBM
+#
+# (C) Copyright IBM Corp. 2019,2020. All Rights Reserved.
```
Comment (npanpaliya) [inline libmemcpy_wrapper/build.sh:6]: Same as above

File: tensorrt-graphsurgeon/meta.yaml [hunk 4]
```diff
@@ -1,9 +1,9 @@
-{% set version = "0.4.1" %}
-{% set sha256 = "03ce0427686e19d60919f6ba68dea5446a26ac53f4398bdea4ec96aec30740e7" %}    #[ppc64le]
-{% set trt_tar = "TensorRT-7.0.0.11.CentOS-7.6.ppc64le-gnu.cuda-10.2.cudnn7.6.tar.gz" %} #[ppc64le]
+{% set version = "0.4.5" %}
```
Comment (npanpaliya) [inline tensorrt-graphsurgeon/meta.yaml:1]: We want these version updates in open-ce/conda_build_config.yaml too.

File: tensorrt-graphsurgeon/meta.yaml [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
+{% set sha256 = "32ea84fb87619ec6b90ca3f4bb9d9dd3d5ca12f6a5f2db9777bfc858d9e62464" %}    #[ppc64le]
+{% set trt_tar = "TensorRT-7.2.0.13.CentOS-8.1.ppc64le-gnu.cuda-11.0.cudnn8.0.tar.gz" %} #[ppc64le]
 
-{% set sha256 = "92fab8aa1de9e5bb6904940e840d553bf64caf39ed5fd6a749297c95e3619051" %}    #[x86_64]
-{% set trt_tar = "TensorRT-7.0.0.11.CentOS-7.6.x86_64-gnu.cuda-10.2.cudnn7.6.tar.gz" %}  #[x86_64]
+{% set sha256 = "3d42890424e2a2a155df67aaace45d001a95f12d6781008c5dcf62ea480ea664" %}   #[x86_64]
+{% set trt_tar = "TensorRT-7.2.1.6.CentOS-7.6.x86_64-gnu.cuda-11.0.cudnn8.0.tar.gz" %}  #[x86_64]
```
Comment (npanpaliya) [inline tensorrt-graphsurgeon/meta.yaml:7]: Do we not want to support RH7 for ppc64le and RH8 for x86?

Comment (npanpaliya) [reply tensorrt-graphsurgeon/meta.yaml:7]: Or NVIDIA doesn't support it?

File: tensorrt-onnx-graphsurgeon/build.sh [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+# *****************************************************************
+#
+# Licensed Materials - Property of IBM
+#
+# (C) Copyright IBM Corp. 2019,2020. All Rights Reserved.
```
Comment (npanpaliya) [inline tensorrt-onnx-graphsurgeon/build.sh:6]: Copyright year to be 2020.

File: tensorrt-onnx-graphsurgeon/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    - python {{ python }}
+
+about:
+  home: https://developer.nvidia.com/tensorrt
+  license: NVIDIA Software License
+  summary: graphsurgeon allows you to transform TensorFlow graphs.
+  description: |
+    graphsurgeon allows you to transform TensorFlow graphs. Its capabilities are broadly divided into two categories - search and manipulation. 
+    Search functions allow you to find nodes in a TensorFlow graph. Manipulation functions allow you to modify, add, or remove nodes.
+  doc_url: https://docs.nvidia.com/deeplearning/sdk/tensorrt-api/python_api/graphsurgeon/graphsurgeon.html
```
Comment (npanpaliya) [inline tensorrt-onnx-graphsurgeon/meta.yaml:38]: I see the same information in about section of graphsurgeon too. Hope this is fine.

File: tensorrt/meta.yaml [hunk 8]
```diff
@@ -1,9 +1,10 @@
-{% set tensorrt_version = "7.0.0.11" %}
+{% set tensorrt_version = "7.2.0.13" %}                                                  #[ppc64le]
 {% set sha256 = "03ce0427686e19d60919f6ba68dea5446a26ac53f4398bdea4ec96aec30740e7" %}    #[ppc64le]
```
Comment (npanpaliya) [inline tensorrt/meta.yaml]: sha256 value isn't changed for ppc?

File: libclock_gettime_wrapper/meta.yaml (hunk 2 above)
Comment (jayfurmanek) [reply libclock_gettime_wrapper/meta.yaml:22]: yes, that needs filled in

File: tensorrt-graphsurgeon/meta.yaml (hunk 4 above)
Comment (jayfurmanek) [reply tensorrt-graphsurgeon/meta.yaml:1]: yes, I have those local - will push those to the open PR there.

File: tensorrt-graphsurgeon/meta.yaml (hunk 5 above)
Comment (jayfurmanek) [reply tensorrt-graphsurgeon/meta.yaml:7]: This is the way they come from Nvidia. You pick the version, then the CUDA level, not the OS level.

File: tensorrt-onnx-graphsurgeon/meta.yaml (hunk 7 above)
Comment (jayfurmanek) [reply tensorrt-onnx-graphsurgeon/meta.yaml:38]: yeah- there are no online docs for `onnx-graphsurgen` yet, so I put in the details for `graphsurgeon`

File: tensorrt/meta.yaml (hunk 8 above)
Comment (jayfurmanek) [reply tensorrt/meta.yaml]: Fixed

File: tensorrt-onnx-graphsurgeon/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,42 @@
+{% set version = "0.2.0" %}                                                              #[ppc64le]
```
Comment (npanpaliya) [inline tensorrt-onnx-graphsurgeon/meta.yaml:1]: Should these versions be read from open-ce/conda_build_config.yaml?

**Comment by npanpaliya** [comment]
@jayfurmanek - We want to keep TRT 7.0 to support cuda 10.2 on ppc64le. So, I think we should have kept TRT 7.0 recipe too along with 7.2. This PR removes 7.0 recipe.

File: tensorrt7.2/tensorrt/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+           - {{ compiler('cxx') }}
+           - cmake {{ cmake }}
+           - make {{ make }}
+         host:
+           - python {{ python }}
+           - protobuf {{ protobuf }}
+           - libprotobuf {{ protobuf }}
+           - cudnn {{ cudnn }}
+           - tensorrt {{ tensorrt_version }}
+           - libmemcpy_wrapper           #[x86_64]
```
Comment (bnemanich) [inline tensorrt7.2/tensorrt/meta.yaml:91]: I don't think the new wrapper libs should be needed, since the tensorrt packages should be pulling them in.

Comment (jayfurmanek) [reply tensorrt7.2/tensorrt/meta.yaml:91]: yea

File: tensorrt7.2/tensorrt/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+  script_env:
+    - CUDA_HOME
+
+outputs:
+   - name: tensorrt
+     script: build-tensorrt.sh
+     requirements:
+         build:
+           - {{ compiler('cxx') }}
+           - cmake {{ cmake }}
```
Comment (npanpaliya) [inline tensorrt7.2/tensorrt/meta.yaml:55]: Pins for make/cmake

File: tensorrt7.2/tensorrt/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+
+       # 03xx - patch temporary to fix a problem that when fixed upstream can be removed
+       - 0301-fix-GLIBC_2.14-link-issue-with-libnvinfer.so.patch # [x86_64]
+       - 0302-Fix-libcaffeparser-name-for-samples.patch
+       - 0304-modified-print-function-syntax-as-per-py3.patch
+       #- add-makefiles-for-new-samples.patch
+    folder: TensorRT
+
+build:
+  number: 3
```
Comment (npanpaliya) [inline tensorrt7.2/tensorrt/meta.yaml:43]: Build number to be reset t to 1.

**Pull request merged**